### PR TITLE
x265: enable HDR10+ parsing

### PR DESCRIPTION
**Pull request by janngobble**
x265: enable HDR10+

#### Description

Allow HDR10+ parsing and usage of json files

Handle request from ticket:
https://trac.macports.org/ticket/66455

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by reneeotten — CHANGES_REQUESTED** [review]
If these changes are deemed useful/acceptable (I'm not using this software so don't know), then this PR will require at least the following changes:
- this change of configure arguments will change the build so you need to increase the "revision" field. 
- the commit message should be reworded to follow the guidelines as stayed in the PR checklist. In short: "<portname>: short description"

**Comment by janngobble** [comment]
> If these changes are deemed useful/acceptable (I'm not using this software so don't know), then this PR will require at least the following changes:
> 
> * this change of configure arguments will change the build so you need to increase the "revision" field.
> * the commit message should be reworded to follow the guidelines as stayed in the PR checklist. In short: ": short description"

Is this what you asked for, @reneeotten? This is my first commit using GitHub this way into this project. I shortened short description, moving the rest into the full description and bumped revision from 0 to 1, then had GitHub commit those changes directly to patch-1's branch.

Thanks!

**Comment by reneeotten** [comment]
@janngobble change to revision is indeed correct. Your commit message is not correct though; there should be only one commit and it should read something like "x265: enable HDR10+"

**Comment by janngobble** [comment]
> @janngobble change to revision is indeed correct. Your commit message is not correct though; there should be only one commit and it should read something like "x265: enable HDR10+"

Okay, so - newbie wrt the GitHub interface - there are already two commits. 1 w/Portfile logic changes, and one with revision. You say there only should be one with “ x265: enable HDR10+” as the commit message.

What’s my next move? Do I find someway to remove the commits and redo it as one commit from scratch? I’m sorry for asking but I thought I followed the rules, and obviously didn’t.

Thanks in advance.

File: multimedia/x265/Portfile [hunk 1]
```diff
@@ -52,6 +52,7 @@ platform darwin 10 {
     # Rosetta build has to override CMAKE_SYSTEM_PROCESSOR: https://trac.macports.org/ticket/64528
     if {${build_arch} eq "ppc"} {
         configure.args-append -DOVERRIDE_SYSTEM_PROCESSOR="ppc"
+        configure.args-append -DENABLE_HDR10_PLUS=On
```
Comment (neverpanic) [inline multimedia/x265/Portfile:55]: Your change enables HDR10_PLUS when

- installing x265 on darwin 10 PowerPC, or
- when installing with the `+highdepth` variant on all platforms

That does not seem to make a lot of sense to me. Would you not only want to have it enabled when the `+highdepth` variant is enabled? If so, this line should be removed.

**Comment by neverpanic** [comment]
> > @janngobble change to revision is indeed correct. Your commit message is not correct though; there should be only one commit and it should read something like "x265: enable HDR10+"
> 
> Okay, so - newbie wrt the GitHub interface - there are already two commits. 1 w/Portfile logic changes, and one with revision. You say there only should be one with “ x265: enable HDR10+” as the commit message.
> 
> What’s my next move? Do I find someway to remove the commits and redo it as one commit from scratch? I’m sorry for asking but I thought I followed the rules, and obviously didn’t.

You'd have to do this on the command line and do what's called an interactive rebase of these two commits, where you then make the revision change commit a `fixup` entry, and the interactive rebase will merge the changes into the previous commit. Then you'd have to force-push the change to your branch.

However, we could probably do the same on the GitHub interface if we use the squash and merge option, which seems easier in this case.

File: multimedia/x265/Portfile (hunk 1 above)
Comment (janngobble) [reply multimedia/x265/Portfile]: I altered that. As I said in the ticket - this was originally done by someone else who had that arg in that location. I have now removed it and left it only in the location enabling +highdepth. I understand why the logic there is wrong.

**Review by kencu — APPROVED** [review]
looks ok to me. We may need to tweak compiler selection as this option requires c++11, by adding this in the highdepth variant.

compiler.cxx_standard 2011

Too bad I never circled back and sorted out how to do the highdepth dance while using muniversal.

**Comment by kencu** [comment]
enabling the new option in the 10 bit and 12 bit builds seems ok… does it do anything though in the default 8 bit build though ? I’d have to see. I set the CI to run so we can see the cmake output.

**Comment by kencu** [comment]
added the c++11 requirement here

debdbeb96873156800e99c3f55b9a72258a63bae

**Comment by i0ntempest** [comment]
Looks like after this change `libx265.dylib` is no longer built so ffmpeg will fail to configure (and installed ffmpeg port will break). `-DENABLE_SHARED=OFF` is already there before this, why does the dylib went missing only after this change?

**Comment by janngobble** [comment]
On both Intel and ARM?

**Comment by janngobble** [comment]
> Looks like after this change `libx265.dylib` is no longer built so ffmpeg will fail to configure (and installed ffmpeg port will break). `-DENABLE_SHARED=OFF` is already there before this, why does the dylib went missing only after this change?

On both Intel and ARM?

**Comment by i0ntempest** [comment]
> > Looks like after this change `libx265.dylib` is no longer built so ffmpeg will fail to configure (and installed ffmpeg port will break). `-DENABLE_SHARED=OFF` is already there before this, why does the dylib went missing only after this change?
> 
> On both Intel and ARM?

I'm on x86, not sure about arm. Maybe take a look at Ventura arm64 build bot (if there is one)?

**Pull request merged**